portfolio/src/main/webapp/script.js

```diff
@@ -16,13 +16,25 @@
  * Adds a random greeting to the page.
  */
 function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+    const greetings =
+        ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+    let happy = false;
+        
+    while(!happy){
+        // Pick a random greeting.
+        const greeting = greetings[Math.floor(Math.random() * greetings.length)];
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+        // Add it to the page.
+        const greetingContainer = document.getElementById('greeting-container');
```

Comment:
This and the adding of text can go outside of the while loop so that each time they click cancel you don't have to re-find the element. In a case like this, there isn't much to look for, but if the page get's more complex, it might be better to only do this search once.

Reply:
Yeah, I guess I should comment what I'm trying to do here. I'm trying to keep the user within the while loop until they're happy with their greeting, and only when they hit confirm break them out of the loop and print the text to the screen. I see what you mean about not wanting to find the element over and over though.

Reply:
This looks better. If you did want to move the `greetingContainer.innerText = greeting;` to outside the while loop, you could initialize the greeting before the while loop like `let greeting=''` and then you will still have the variable in the scope you need it for, but that isn't necessary because a small conditional check like that isn't costly.